rewriter/SourceRewriter.cpp

```diff
@@ -798,6 +846,52 @@ int main(int argc, const char **argv) {
                        options_parser.getSourcePathList());
   CompilationDatabase &comp_db = options_parser.getCompilations();
 
+  for (auto s : options_parser.getSourcePathList()) {
+    assert(s.starts_with(RootDirectory));
+    std::string relative_path = s.substr(RootDirectory.length());
+    std::string new_path = OutputDirectory;
+    new_path.append(relative_path);
+
+    std::string parent_dir = new_path;
+    auto n = parent_dir.rfind('/');
+    parent_dir = parent_dir.substr(0, n);
+    if (!opendir(parent_dir.c_str())) {
+      auto cmd = "mkdir -p "s + parent_dir;
+      system(cmd.c_str());
+    }
+
+    // Copy the input sources to the output directory
+    std::ifstream src(s, std::ios::binary);
+    std::ofstream dst(new_path, std::ios::binary);
+    dst << src.rdbuf();
+  }
+
+  std::set<Filename> copied_headers;
+  auto asts = std::vector<std::unique_ptr<clang::ASTUnit>>();
+  tool.buildASTs(asts);
+  for (auto &ast : asts) {
+    auto &sm = ast->getSourceManager();
+    for (auto it = sm.fileinfo_begin(); it != sm.fileinfo_end(); it++) {
```

Comment:
Won't this iterate over all sources, not just headers? Do we need the copy above?

Reply:
Right I believe we can remove the loop above.